plugins/module_utils/version.py

```diff
@@ -10,4 +10,41 @@
 __metaclass__ = type
 
 
+import operator
+from functools import wraps
+
 from ansible.module_utils.compat.version import LooseVersion  # noqa: F401, pylint: disable=unused-import
+
+
+def _version_compare(op):
+    @wraps(op)
+    def _op(a, b):
+        return op(LooseVersion(a), LooseVersion(b))
+    return _op
+
+
+version_ops = {
+    '<=': _version_compare(operator.le),
+    '>=': _version_compare(operator.ge),
+    '<': _version_compare(operator.lt),
+    '>': _version_compare(operator.gt),
+    '==': _version_compare(operator.eq),
+    '!=': _version_compare(operator.ne),
+}
+
+
+def find_op(name):
+    if name is None:
+        return None, None, None
+    for op in version_ops:
+        if op in name:
```

Comment:
I'm wondering whether the function should check whether other operators also appear in the name, so if someone specifies `foo != 3 < 5` you get some better results.

Reply:
not sure if I got the idea here, you mean more than on operator per entry?

Reply:
For example `name` being `foo != 3 < 5`. There are two operators here: `!=` and `<`.

Reply:
Check also https://packaging.python.org/en/latest/specifications/version-specifiers/, in particular https://packaging.python.org/en/latest/specifications/version-specifiers/#id5.